libraries/AP_Compass/AP_Compass.h

```diff
@@ -294,12 +296,18 @@ friend class AP_Compass_Backend;
         Vector3f field[COMPASS_MAX_INSTANCES];
     } _hil;
 
+    //singleton get instance function
+    static Compass& get_frontend(void);
+ 
 private:
```

Comment:
see comment above. The only thing that should actually need to be static is the _s_instance. not every method in the class.